### DISCUSSION TITLE
Getting the original markdoc post/content back from an AST

### DISCUSSION THREAD

**Discussion by Aldo111** [markdoc/markdoc#39] [Q&A] [answered]
Hey, I was just curious if there was an easy/built-in way to get back the original markdoc code/content from a parsed AST (or if this is something that's not an intended feature).

I was just bouncing around the idea of storing the AST in a database for posts instead of the original string/post and so was curious if there was a way to go back to the original string for editing purposes without manually walking the tree.

I'm guessing it might be a bad idea to store the original AST (in case grammar/syntax/structure changes in future versions?) but was curious in general.

**Comment by mfix-stripe** [comment, marked as answer]
@Aldo111 we don't think this is a bad idea at all! This is actually something we have long wanted to do (@rpaul-stripe @matv-stripe).

There are currently some limitations that make it difficult to reproduce the exact document (e.g. annotations), but with some improvements, this is definitely possible and something that should exist at some point. 

The good news is, the Markdoc document renderer shouldn't require any changes to core: it should just be a pure function that takes in an AST and returns a string containing the Markdoc document.

**Reply by Aldo111** [reply]
That's awesome to hear. I really dig markdocs, it's very easy to work with and customized, and super pumped about the direction it's going in! Keep up the great work! :bufo-ty:
